packages/block-library/src/query-pagination/edit.js

```diff
@@ -30,24 +31,30 @@ const ALLOWED_BLOCKS = [
 ];
 
 export default function QueryPaginationEdit( {
-	attributes: { paginationArrow, showLabel },
+	attributes: { paginationArrow, showLabel, midSize },
 	setAttributes,
 	clientId,
 } ) {
-	const hasNextPreviousBlocks = useSelect( ( select ) => {
-		const { getBlocks } = select( blockEditorStore );
-		const innerBlocks = getBlocks( clientId );
-		/**
-		 * Show the `paginationArrow` and `showLabel` controls only if a
-		 * `QueryPaginationNext/Previous` block exists.
-		 */
-		return innerBlocks?.find( ( innerBlock ) => {
-			return [
-				'core/query-pagination-next',
-				'core/query-pagination-previous',
-			].includes( innerBlock.name );
-		} );
-	}, [] );
+	const { getBlocks } = useSelect( ( select ) => select( blockEditorStore ) );
+	const innerBlocks = getBlocks( clientId );
+	/**
+	 * Show the `paginationArrow` and `showLabel` controls only if a
+	 * `QueryPaginationNext/Previous` block exists.
+	 */
+	const hasNextPreviousBlocks = innerBlocks?.find( ( innerBlock ) =>
+		[
+			'core/query-pagination-next',
+			'core/query-pagination-previous',
+		].includes( innerBlock.name )
+	);
+	/**
+	 * Show the `midSize` control only if a
+	 * `QueryPaginationNumbers` block exists.
+	 */
+	const hasPageNumbersBlock = innerBlocks?.find( ( innerBlock ) =>
```

Comment:
>Or do you mean the midSize parameter, that this should be transferred to the Page Numbers block instead of Query Pagination block?

This is what I meant. Move the attribute and the control to `Page Numbers` block, since it only applies to that one and I don't think it's common to have more `Page Number` blocks in the same pagination to have them in sync.

The difference here is that in a pagination we usually have both `next and previous` and that's why we decided to sync them(ex. `arrow`) and handle that in the parent block.

@jameskoster do you think this control should be in the block toolbar or in the inspector controls? Any thoughts?

Reply:
+1 to moving the setting to the Page Numbers block.

The Inspector feels like the correct place for this setting, but the control could use a little polishing – it's a bit tricky for users to understand without any prior knowledge of `midSize` (which most will not have). The disconnect between the value and the number of links is a little uncanny (IE a value of 2 produces 4 links). Similarly increasing the value by one produces _two_ more links.

I'd be inclined to simplify this by just offering a range control. It still manipulates `midSize` but obfuscates the math:

<img width="998" alt="number of links" src="https://github.com/WordPress/gutenberg/assets/846565/128cff71-dc25-4f38-b4df-92f6131080e7">

What do you think?

Reply:
Thank you for your feedback! I updated my code and moved the control to the Query Pagination Numbers block as a RangeControl.